openfl/component/aggregation_functions/weighted_average.py

```diff
@@ -15,6 +15,15 @@ def weighted_average(tensors, weights):
 class WeightedAverage(AggregationFunctionInterface):
     """Weighted average aggregation."""
 
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        """Use the singleton instance if it has already been created."""
+        if not isinstance(cls._instance, cls):
+            cls._instance = object.__new__(cls, *args, **kwargs)
+
+        return cls._instance
```

Comment:
It might be better to implement the Singleton metaclass so that it can easily add to any class. What do you think?